webapp/src/components/viewHeader/viewHeaderSearch.tsx

```diff
@@ -14,21 +14,17 @@ type Props = {
 }
 
 const ViewHeaderSearch = (props: Props) => {
-    const {boardTree} = props
+    const {boardTree, setSearchText} = props
     const intl = useIntl()
 
     const searchFieldRef = useRef<{focus(selectAll?: boolean): void}>(null)
-    const [isSearching, setIsSearching] = useState(Boolean(props.boardTree.getSearchText()))
+    const [isSearching, setIsSearching] = useState(Boolean(boardTree.getSearchText()))
     const [searchValue, setSearchValue] = useState(boardTree.getSearchText())
 
     useEffect(() => {
         searchFieldRef.current?.focus()
     }, [isSearching])
 
-    useEffect(() => {
-        setSearchValue(boardTree.getSearchText())
-    }, [boardTree])
-
```

Comment:
Note:

I removed this because this did nothing. Search doesn't get triggered if you don't do `setSearchText` and setting this would just temporarily set state but wouldn't actually do anything. 

Feel free to correct me if I'm wrong but I came to conclusion we don't need this.

Reply:
This seems to be assigned the changed value coming from `Editable` back to `Editable`. But I see we're doing this everywhere we use `Editable`, so there could be an edge case this is covering. @jespino would know best.

I think we should be careful removing this.

Reply:
Note to self:

Add reason why we keep this part in the code.

Reply:
I think it is ok, lets merge it and see where it breaks. clearly makes more sense to remove that block than keeping it. And the new change on the incrementalUpdate, I think is correct from the data perspective, but I would like to investigate it a bit more later.